src/v/cloud_storage_clients/tests/xml_sax_parser_test.cc

```diff
@@ -247,3 +273,20 @@ BOOST_AUTO_TEST_CASE(test_parse_abs_with_continuation) {
     BOOST_REQUIRE_EQUAL(result.is_truncated, true);
     BOOST_REQUIRE_EQUAL(result.next_continuation_token, "nnn");
 }
+
+BOOST_AUTO_TEST_CASE(test_parse_abs_with_blob_prefix) {
+    cloud_storage_clients::xml_sax_parser p{};
+    ss::temporary_buffer<char> buffer(
+      abs_payload_with_blob_prefix.data(), abs_payload_with_blob_prefix.size());
+
+    p.start_parse(std::make_unique<cloud_storage_clients::abs_parse_impl>());
+    p.parse_chunk(std::move(buffer));
+    p.end_parse();
+
+    auto result = p.result();
+    BOOST_REQUIRE(result.contents.empty());
+    BOOST_REQUIRE_EQUAL(result.common_prefixes.size(), 1);
+    BOOST_REQUIRE_EQUAL(
+      result.common_prefixes[0],
+      "cluster_metadata/bb7527f1-3227-4d55-86da-c133ec955ea9/manifests/2/");
+}
```

Comment:
if i remove the first two commits this test still passes. so it seems like either the test isn't isn't reproducing the issue, or a reproducer test wasn't added to the pr.

Reply:
Oops! Thanks for the callout, my XML was slightly off and had confirmed my end to end testing was fixed with the changes before writing the test.

Force pushed the correct XML for the test case, confirmed that it fails before and passes after my fix.

Reply:
```diff
 <Blobs>
  <Blobs>
    <BlobPrefix>
    <BlobPrefix>
      <Name>cluster_metadata/bb7527f1-3227-4d55-86da-c133ec955ea9/manifests/2/</Name>
      <Name>cluster_metadata/bb7527f1-3227-4d55-86da-c133ec955ea9/manifests/2/</Name>
+      <Properties>
	<Creation-Time>Thu, 25 Jul 2024 14:07:26 GMT</Creation-Time>
	<Creation-Time>Thu, 25 Jul 2024 14:07:26 GMT</Creation-Time>
	<Last-Modified>Thu, 25 Jul 2024 14:07:26 GMT</Last-Modified>
	<Last-Modified>Thu, 25 Jul 2024 14:07:26 GMT</Last-Modified>
	<Etag>0x8DCACB31CE7DB5C</Etag>
	<Etag>0x8DCACB31CE7DB5C</Etag>
@@ -178,6 +179,7 @@ static constexpr std::string_view abs_payload_with_blob_prefix = R"XML(
	<LeaseStatus>unlocked</LeaseStatus>
	<LeaseStatus>unlocked</LeaseStatus>
	<LeaseState>available</LeaseState>
	<LeaseState>available</LeaseState>
	<ServerEncrypted>true</ServerEncrypted>
	<ServerEncrypted>true</ServerEncrypted>
+      </Properties>
    </BlobPrefix>
    </BlobPrefix>
  </Blobs>
  </Blobs>
  <NextMarker />
  <NextMarker />
```